### PR TITLE
feat: filter query preview

### PR DESCRIPTION
**Pull request by lauramargar**
# Pull request template
 
It would be great if we could add filters & extra params to the search request, now extra params are developed, so we also have to add filters. These params could help to narrow the results, that could be very interesting for some customers.

In this PR, I have merged the branch of [filter carrousels of brand recommendations](https://github.com/empathyco/x/pull/1292). So we should have the complete functionality of filter query previews.

## Motivation and context

- [ ] Dependencies. If any, specify:
- [X] Open issue. If applicable, link: [EMP-1825](https://searchbroker.atlassian.net/browse/EMP-1825)

## Type of change

- [ ] Bug fix (non-breaking change that fixes an issue)
- [X] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that causes existing functionality to not work as expected)
- [ ] Change requires a documentation update

## What is the destination branch of this PR?

- [X] `Main`
- [ ] Other. Specify:

## How has this been tested?

Here we can test if brand recommendation carrousels are filtered and if when we click the query preview button, filters are applied.

[EMP-1825]: https://searchbroker.atlassian.net/browse/EMP-1825?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: packages/x-components/src/x-modules/queries-preview/components/query-preview-button.vue
Comment (CachedaCodes) [inline packages/x-components/src/x-modules/queries-preview/components/query-preview-button.vue:1]: Update the tests of `QueryPreviewButton` to take the filters into account.

File: packages/x-components/src/x-modules/facets/wiring.ts [hunk 1]
```diff
@@ -4,9 +4,23 @@ import { createRawFilters } from '../../utils/filters';
...
+ * @internal
+ */
+const moduleName = 'facets';
+
+/**
+ * WireCommit for {@link SearchXModule}.
+ *
+ * @internal
+ */
+const wireCommitModule = namespacedWireCommit(moduleName);
```
Comment (CachedaCodes) [inline packages/x-components/src/x-modules/facets/wiring.ts:22]: The `wireCommit` that the wiring in the module uses should be replaced by a `namespacedWireCommitWithoutPayload`.

The wires that commit mutations with payload should keep using the one you created here.

**Review by CachedaCodes — CHANGES_REQUESTED** [review]

File: packages/x-components/src/x-modules/url/wiring.ts [hunk 2]
```diff
@@ -1,5 +1,6 @@
 import { namespacedWireCommit } from '../../wiring/namespaced-wires.factory';
 import { createWiring } from '../../wiring/wiring.utils';
+import { createRawFilters } from '../../utils/index';
```
Comment (CachedaCodes) [inline packages/x-components/src/x-modules/url/wiring.ts:3]:
```suggestion
import { createRawFilters } from '../../utils/filters';
```

File: packages/x-components/src/x-modules/facets/wiring.ts [hunk 3]
```diff
@@ -1,12 +1,32 @@
...
 import { DefaultFacetsService } from './service/facets.service';
 
+/**
+ * `facets` {@link XModuleName | XModule name}.
+ *
+ * @internal
+ */
+const moduleName = 'facets';
+
+const wireCommitWithoutPayload = namespacedWireCommitWithoutPayload(moduleName);
```
Comment (CachedaCodes) [inline packages/x-components/src/x-modules/facets/wiring.ts:21]: Document this and put it below the `wireCommit`.

**Review by CachedaCodes — CHANGES_REQUESTED** [review]

**Review by CachedaCodes — APPROVED** [review]

**Review by annacv — APPROVED** [review]
gj!

**Pull request merged**